### DISCUSSION TITLE
`ScaleShiftMACE`: `torch.autograd.grad()` `outputs` parameter using `inter_e` instead of `total_energy`?

### DISCUSSION THREAD

**Discussion by JSLJ23** [ACEsuit/mace#345] [Q&A] [answered]
I have a question behind the logic / reasoning of how the forces are computed in the `ScaleShiftMACE` variant of MACE.
I notice in the `get_outputs()` function, the energy term passed to this function is the `inter_e` which is then used by the `compute_forces()` function to call `torch.autograd.grad()` for the gradients with respect to the positions.
I have 2 questions regarding this:

1. What does this mean physically when the `inter_e` is used instead of the `total_energy`?
2. Is there a physical meaning to the `inter_e` and why back-propagating with it in the context of `ScaleShiftMACE` would lead to forces?

**Comment by pobo95** [comment]
To fit a potential to the binding energy.

`inter_e` is `E0` (energy of isolated atoms).

**Comment by ilyes319** [comment, marked as answer]
The `inter_e` corresponds to energy of the system without the isolated energy. This quantity is very physical, as isolated energies are just arbitrary shifts that does not depend on the geometry. The reason why we backpropagate through it, is to avoid floating point precision issues, as the total energy is usually a large number for most DFT codes.

**Comment by JSLJ23** [comment]
Thank you for your answer. Could you expand a bit more on the "isolated energies" you were referring to previously? Does it refer to the single point energy of an isolated atom?

**Reply by ilyes319** [reply]
Yes exactly.
